spec/controllers/migrations/external_controller_spec.rb

```diff
@@ -83,6 +83,250 @@
+      describe 'because payload encapsulation is not conform' do
+        it 'returns :bad_request without creating a user' do
+          encrypted_params = encrypt_params({ user: valid_user_params }, expires_in: 10.seconds)
+          expect_any_instance_of(described_class).not_to receive(:generate_secure_pwd)
+          expect { post :create_user, params: { not_v2: { encrypted_params: } } }.not_to change(User, :count)
+          expect(ActionMailer::MailDeliveryJob).not_to have_been_enqueued
+          expect(response).to have_http_status(:bad_request)
+        end
+      end
+
+      describe 'because the encrypted_params cipher isn\'t a String' do
+        it 'returns :bad_request without creating a user' do
+          expect_any_instance_of(described_class).not_to receive(:generate_secure_pwd)
+          expect { post :create_user, params: { v2: { encrypted_params: { something: 'else' } } } }.not_to change(User, :count)
+          expect(ActionMailer::MailDeliveryJob).not_to have_been_enqueued
+          expect(response).to have_http_status(:bad_request)
+        end
+      end
+
+      describe 'because the decrypted params isn\'t a Hash' do
+        it 'returns :bad_request without creating a user' do
+          encrypted_params = encrypt_params('I am  a Hash!!', expires_in: 10.seconds)
+          expect_any_instance_of(described_class).not_to receive(:generate_secure_pwd)
+          expect { post :create_user, params: { v2: { encrypted_params: } } }.not_to change(User, :count)
+          expect(ActionMailer::MailDeliveryJob).not_to have_been_enqueued
+          expect(response).to have_http_status(:bad_request)
+        end
+      end
+
+      describe 'because the encrypted payload expired' do
+        it 'returns :bad_request without creating a user' do
+          encrypted_params = encrypt_params({ user: valid_user_params }, expires_in: 10.seconds)
+          travel_to 11.seconds.from_now
+          expect_any_instance_of(described_class).not_to receive(:generate_secure_pwd)
+          expect { post :create_user, params: { v2: { encrypted_params: } } }.not_to change(User, :count)
+          expect(ActionMailer::MailDeliveryJob).not_to have_been_enqueued
+          expect(response).to have_http_status(:bad_request)
+        end
+      end
+
+      describe 'because the ciphertext was not generated with the same configuration' do
+        it 'returns :bad_request without creating a user' do
+          key = Rails.application.secrets.secret_key_base[1..32]
+
+          encrypted_params = encrypt_params({ user: valid_user_params }, key:, expires_in: 10.seconds)
+          expect_any_instance_of(described_class).not_to receive(:generate_secure_pwd)
+          expect { post :create_user, params: { v2: { encrypted_params: } } }.not_to change(User, :count)
+          expect(ActionMailer::MailDeliveryJob).not_to have_been_enqueued
+          expect(response).to have_http_status(:bad_request)
+        end
+      end
+    end
+  end
+
```

Comment:
I've tried to make good examples following some the guidelines of https://www.betterspecs.org/
However, we can dry these examples out.
Tell me what you think guys,